File: NAP II - POO/Python/UFRAcommerce/item.py
class Item:
    def __init__(self, quantidade, produto):
        self.quantidade = quantidade
        self.produto = produto
        self.subtotal = self.calcularSubtotal()

    def calcularSubtotal(self):
        return self.quantidade * self.produto.preco

    def setQuantidade(self, quantidade):
        self.quantidade = quantidade
        self.subtotal = self.calcularSubtotal()

    def __str__(self):
        return f"Item(quantidade={self.quantidade}, produto={self.produto.nome}, subtotal={self.subtotal})"

File: NAP_II_JAVA/ClasseProdutoComboEItem/Item.java
public class Item {
    private int quantidade;
    private double subtotal;
    private Produto produto; // Produto que representa este item

    // Construtor não padrão
    public Item(int quantidade, Produto produto) {
        this.quantidade = quantidade;
        this.produto = produto;
        calcularSubtotal(); // Calcula o subtotal automaticamente ao criar o item
    }

    // Getters e Setters
    public int getQuantidade() {
        return quantidade;
    }

    public void setQuantidade(int quantidade) {
        this.quantidade = quantidade;
    }

    public double getSubtotal() {
        return subtotal;
    }

    public Produto getProduto() {
        return produto;
    }

    public void setProduto(Produto produto) {
        this.produto = produto;
    }
// Sobrescrevendo toString
    @Override
    public String toString() {
    return "Item{" +
            "quantidade=" + quantidade +
            ", subtotal=" + subtotal +
            ", produto=" + produto +
            '}';
    }
}

